cis_policies_generator/src/index.ts

```diff
@@ -131,29 +156,78 @@ function parseBenchmarks(folder: string): BenchmarkSchema[] {
     });
 }
 
-function generateOutputFiles(benchmarks: BenchmarkSchema[]): void {
-    const result: any = {
-        "policies": {}
-    };
-
+function generateRulesMetadataFiles(benchmarks: BenchmarkSchema[]): void {
     for (const benchmark of benchmarks) {
         console.log("Parsed total of", benchmark.rules.length, "rules in benchmark", benchmark.filename);
-        result.policies[benchmark.filename] = {};
+        const benchmark_id = getBenchmarkAttr(benchmark.metadata, "id")
         for (let rule of benchmark.rules) {
-            result.policies[benchmark.filename][rule.rule_number] = rule;
+            const ruleNumber = rule.rule_number!.replaceAll(".", "_");
+            const rule_folder = `../bundle/compliance/${benchmark_id}/rules/cis_${ruleNumber}`
+            const metadata_file = rule_folder + "/data.yaml";
+            delete rule.rule_number;
+
+            if (fs.existsSync(rule_folder)) {
+                rule.default_value = getExistingDefaultVal(metadata_file);
+                fs.writeFileSync(metadata_file, YAML.stringify({metadata: rule} as MetadataFile));
+            }
         }
-        fs.writeFileSync(output_folder + "/" + benchmark.filename + ".yaml", YAML.stringify(benchmark.rules));
     }
-    fs.writeFileSync(output_folder + "/" + config.get("output_filename"), YAML.stringify(result));
 }
 
+function constructRuleTags(benchmark_metadata: BenchmarkMetadata, rule_number: string, section: string): string[] {
+    const benchmark_type = getBenchmarkAttr(benchmark_metadata, "tag")
+    return ["CIS", benchmark_type, "CIS " + rule_number, section].filter(Boolean)
+}
+
+function getBenchmarkAttr(benchmark_metadata: BenchmarkMetadata, field: string): string {
+    for (let [type, attr] of Object.entries(BENCHMARK_TYPES)) {
+        if (benchmark_metadata.name.includes(type)) {
+            // @ts-ignore
+            return attr[field];
+        }
+    }
+
+    return ""
+}
+
+function getK8sVersionFromBenchmark(filename: string): string {
+    if (filename.includes("CIS_Kubernetes")) {
+        // We count on the third element in the k8s benchmark filename to be the k8s version.
+        return filename.split("_")[2].toLowerCase()
+    }
+
+    return ""
+}
+
+function generateBenchmarkId(metadata: BenchmarkMetadata) {
```

Comment:
Not every benchmark we will have is related to Kubernetes.
Benchmark Id should be
`CIS_Amazon_Elastic_Kubernetes_Service_(EKS)_Benchmark_v1.1.0`
or
`CIS_Kubernetes_V1.23_Benchmark_v1.0.1`

Reply:
OK, I'll use the file name.